src/Microsoft.Azure.WebJobs.Logging/FunctionInstanceLogItem.cs

```diff
@@ -46,17 +46,38 @@ public class FunctionInstanceLogItem : IFunctionInstanceBaseEntry
         /// <summary>If set, the time the function completed (either successfully or failure). </summary>
         public DateTime? EndTime { get; set; }
 
+        /// <summary>A function outputs a heartbeat while it's running. 
+        /// If EndTime = null and heartbeat is "old", then the function is likley abandoned (machine killed).
+        /// It's always possible a node loses network connectity, and so we appear abandoned, but 
+        /// the function successfully completes</summary>
+        /// <inheritdoc/>
+        public DateTime? HeartbeatExpireTime { get; set; }
+
         /// <summary>
-        /// Current status of this instance. 
+        /// Given log item a chance to refresh. 
         /// </summary>
-        public FunctionInstanceStatus Status { get; set; }
+        public virtual void Refresh()
+        {
+            this.HeartbeatExpireTime = DateTime.UtcNow.AddMinutes(50);
```

Comment:
will change to pass this  in. 

---
In reply to: [90368373](https://github.com/Azure/azure-webjobs-sdk/pull/928#discussion_r90368373) [](ancestors = 90368373)